Examples/Python/src/ExampleAlgorithms.cpp

```diff
@@ -32,92 +32,24 @@ void addExampleAlgorithms(Context& ctx) {
-    ACTS_PYTHON_MEMBER(outputParticlesFinal);
-    ACTS_PYTHON_MEMBER(outputSimHits);
-    ACTS_PYTHON_MEMBER(imputParametrisationNuclearInteraction);
-    ACTS_PYTHON_MEMBER(randomNumbers);
-    ACTS_PYTHON_MEMBER(trackingGeometry);
-    ACTS_PYTHON_MEMBER(magneticField);
-    ACTS_PYTHON_MEMBER(pMin);
-    ACTS_PYTHON_MEMBER(emScattering);
-    ACTS_PYTHON_MEMBER(emEnergyLossIonisation);
-    ACTS_PYTHON_MEMBER(emEnergyLossRadiation);
-    ACTS_PYTHON_MEMBER(emPhotonConversion);
-    ACTS_PYTHON_MEMBER(generateHitsOnSensitive);
-    ACTS_PYTHON_MEMBER(generateHitsOnMaterial);
-    ACTS_PYTHON_MEMBER(generateHitsOnPassive);
-    ACTS_PYTHON_MEMBER(averageHitsPerParticle);
-    ACTS_PYTHON_STRUCT_END();
-  }
-
-  {
-    using Alg = ActsExamples::ParticlesPrinter;
-
-    auto alg =
-        py::class_<Alg, ActsExamples::BareAlgorithm, std::shared_ptr<Alg>>(
-            mex, "ParticlesPrinter")
-            .def(py::init<const Alg::Config&, Acts::Logging::Level>(),
-                 py::arg("config"), py::arg("level"));
-
-    py::class_<Alg::Config>(alg, "Config")
-        .def(py::init<>())
-        .def_readwrite("inputParticles", &Alg::Config::inputParticles);
-  }
-
-  {
-    using Alg = ActsExamples::HitsPrinter;
-
-    auto alg =
-        py::class_<Alg, ActsExamples::BareAlgorithm, std::shared_ptr<Alg>>(
-            mex, "HitsPrinter")
-            .def(py::init<const Alg::Config&, Acts::Logging::Level>(),
-                 py::arg("config"), py::arg("level"));
-
-    auto c = py::class_<Alg::Config>(alg, "Config").def(py::init<>());
-
-    ACTS_PYTHON_STRUCT_BEGIN(c, Alg::Config);
-    ACTS_PYTHON_MEMBER(inputClusters);
-    ACTS_PYTHON_MEMBER(inputMeasurementParticlesMap);
-    ACTS_PYTHON_MEMBER(inputHitIds);
-    ACTS_PYTHON_MEMBER(selectIndexStart);
-    ACTS_PYTHON_MEMBER(selectIndexLength);
-    ACTS_PYTHON_MEMBER(selectVolume);
-    ACTS_PYTHON_MEMBER(selectLayer);
-    ACTS_PYTHON_MEMBER(selectModule);
-    ACTS_PYTHON_STRUCT_END();
-  }
-
-  {
-    using Alg = ActsExamples::TrackParametersPrinter;
-
-    auto alg =
-        py::class_<Alg, ActsExamples::BareAlgorithm, std::shared_ptr<Alg>>(
-            mex, "TrackParametersPrinter")
-            .def(py::init<const Alg::Config&, Acts::Logging::Level>(),
-                 py::arg("config"), py::arg("level"));
-
-    py::class_<Alg::Config>(alg, "Config")
-        .def(py::init<>())
-        .def_readwrite("inputTrackParameters",
-                       &Alg::Config::inputTrackParameters);
-  }
+  ACTS_PYTHON_DECLARE_ALGORITHM(
+      ActsExamples::FatrasSimulation, mex, "FatrasSimulation", inputParticles,
+      outputParticlesInitial, outputParticlesFinal, outputSimHits,
+      imputParametrisationNuclearInteraction, randomNumbers, trackingGeometry,
+      magneticField, pMin, emScattering, emEnergyLossIonisation,
+      emEnergyLossRadiation, emPhotonConversion, generateHitsOnSensitive,
+      generateHitsOnMaterial, generateHitsOnPassive, averageHitsPerParticle);
```

Comment:
That's the only thing I'm skeptical about: this is hard to read, and might make diffs annoying if properties get added. I think the benefits outweigh this downside though.

Reply:
so in principle we *could* provide another `.clang-format` in the `Examples/Python/src` directory that breaks at each comma... I think clang-format does take always the nearest one, right?